packages/concerto-core/test/introspect/modelfile.js

```diff
@@ -608,6 +608,92 @@ describe('ModelFile', () => {
 
     });
 
+    describe('#aliasedImport', () => {
+
+        beforeEach(()=>{
+            modelManager.enableAliasedType=true;
+        });
+        it('should resolve aliased name of import type', () => {
+            const model = `
+            namespace org.acme
+            import org.saluja.{doc as d}`;
+
+            let modelFile = ParserUtil.newModelFile(modelManager, model);
+            modelFile.resolveImport('d').should.equal('org.saluja.doc');
+        });
+
+        it('should not throw if an aliased import exists for a type that exists in a valid namespace', () => {
+            const model1 = `
+            namespace org.saluja.ext
+            asset MyAsset2 identified by assetId {
+                o String assetId
+            }`;
+            const model2 = `
+            namespace org.acme
+            import org.saluja.ext.{MyAsset2 as m}
+            asset MyAsset identified by assetId {
+                o String assetId
+                o m[] arr
+            }`;
+            let modelFile1 = ParserUtil.newModelFile(modelManager, model1);
+            modelManager.addModelFile(modelFile1);
+            let modelFile2 = ParserUtil.newModelFile(modelManager, model2);
+            (() => modelFile2.validate()).should.not.throw();
+        });
+
+        it('should not throw if an duplicate types aliased to distinct aliased names', () => {
+            const model1 = `
+            namespace org.saluja
+            asset MyAsset identified by assetId {
+                o String assetId
+            }`;
+            const model2 = `
+            namespace org.acme
+            asset MyAsset identified by assetId {
+                o String assetId
+            }`;
+
+            const model3 = `
+            namespace org.acme2
+            import org.saluja.{MyAsset as m1}
```

Comment:
question: should it throw if a type from a namespace has already been aliased?
e.g.
```js
import org.saluja.{MyAsset as m1, MyAsset as m2}
```

Reply:
It won't throw in the current implementation. It would just map Myasset to m2 .

Without aliasing also. If we had something like 
```js
import org.saluja.{MyAsset,Myasset}
```

**This would still not throw an error.**


Although if we change this now then we have to change for non-aliased types as well.

Reply:
I can't see anything wrong with either scenario (I'd be happy to be corrected though!), although admittedly they look a little strange. 

Let's keep the current behaviour, but can you open an issue for us to come back to later?

Reply:
raised an issue for above #870